utils.groovy

```diff
@@ -199,4 +199,29 @@ def get_push_trigger() {
     ]
 }
 
+def get_artifacts(pipecfg, stream, basearch) {
+    // Get artifacts from config.yaml
+    def artifacts
+    if  (pipecfg.streams."${stream}".artifacts?."${basearch}" != null) {
+        artifacts = pipecfg.streams."${stream}".artifacts."${basearch}"
+    }
+    else { // Merge default with additional artifacts
+        def default_artifacts =  pipecfg.default_artifacts."${basearch}"
+        def additional_atifacts = pipecfg.streams."${stream}".additional_artifacts?."${basearch}"
+        artifacts = default_artifacts + additional_atifacts
+    }
+    // For all architectures we need to build the metal/metal4k artifacts and the Live ISO. Since the
+    // ISO depends on the Metal/Metal4k images we'll make sure to put the Metal* ones in the first run
+    // and the Live ISO in the second run for parallel runs where the list is split
```

Comment:
It's a bit awkward to do this in this function because it's an implementation detail of the caller that the list then gets split into two lists. I guess it's OK, but let's maybe reword the comment so it says something like: "...put the metal* ones at the start of the list and the live ISO at the end".

Reply:
I agree. I almost think we could have a separate function that does this that is documented appropriately. Is it possible to return multiple values in java/groovy? If so we could just parallel what gets returned by the new function.